src/Voron/StorageEnvironment.cs

```diff
@@ -846,24 +851,29 @@ public void Dispose()
         }
 
         public event Action<LowLevelTransaction> NewTransactionCreated;
-        public event Action<LowLevelTransaction> AfterCommitWhenNewReadTransactionsPrevented;
+        public event Action<LowLevelTransaction> AfterCommitWhenNewTransactionsPrevented;
         internal void TransactionAfterCommit(LowLevelTransaction tx)
         {
             if (ActiveTransactions.Contains(tx) == false)
+            {
+                if (tx.Committed && tx.FlushedToJournal)
```

Comment:
How can that *not* be the case?

Reply:
If you do a transaction with no changes:

```
using (var tx = Env.WriteTransaction())
{
	tx.Commit();
}
```

then `tx.Committed` is `true` but `tx.FlushedToJournal` is `false`

Reply:
So why do you check `tx.Committed` here?

Reply:
To be consistent with the check that we have below, that decides if we should update the transaction counter:

```
if (tx.Committed && tx.FlushedToJournal)
   Interlocked.Exchange(ref _transactionsCounter, tx.Id);
```

`tx.Committed` is supposed to be always `true` at this point. Indeed it looks that checking it is redundant but I don't want to introduce such changes in this kind of PR. I prefer to stay with the redundant check for the sake of safety.

Reply:
Ok, make sense